src/os/ui/search/featureresultcard.js

```diff
@@ -78,18 +97,15 @@ os.ui.search.FeatureResultCardCtrl.prototype.disposeInternal = function() {
   os.ui.search.FeatureResultCardCtrl.base(this, 'disposeInternal');
 
   var mm = os.MapContainer.getInstance();
-  var searchLayer = /** @type {os.layer.Vector} */ (mm.getLayer(os.MapContainer.SEARCH_ID));
 
-  mm.removeFeature(this.feature, undefined, searchLayer);
+  ol.events.unlisten(this.layer.getSource(), goog.events.EventType.PROPERTYCHANGE, this.onSourceChange_, this);
 
-  if (searchLayer) {
-    var source = /** @type {os.source.Vector} */ (searchLayer.getSource());
-    ol.events.unlisten(source, goog.events.EventType.PROPERTYCHANGE, this.onSourceChange_, this);
-    var features = source.getFeatures();
-    if (features.length === 0) {
-      searchLayer.setRemovable(true);
-      mm.removeLayer(os.MapContainer.SEARCH_ID);
-    }
+  var source = this.layer.getSource();
+  source.removeFeature(this.feature);
+
+  if (!source.getFeatures().length) {
+    this.layer.setRemovable(true);
+    mm.removeLayer(os.ui.search.FeatureResultCardCtrl.SEARCH_LAYER_ID);
```

Comment:
This works fine, but if you have a reference to the layer already in `this.layer` you may as well provide it here to avoid the lookup.